server/src/test/java/io/spine/examples/chatspn/server/e2e/given/TestUserEnv.java

```diff
@@ -72,23 +75,34 @@ public static CreatePersonalChat createPersonalChat(UserId creator, UserId membe
                 .newBuilder()
                 .setId(ChatId.generate())
                 .setCreator(creator)
+                .setCreatorName(randomString())
                 .setMember(member)
+                .setMemberName(randomString())
                 .vBuild();
         return command;
     }
 
-    public static ChatPreview chatPreview(CreatePersonalChat command) {
-        var view = PersonalChatView
+    public static ChatCard chatCard(CreatePersonalChat command, UserId cardOwner) {
+        var chatCardId = ChatCardId
                 .newBuilder()
-                .setCreator(command.getCreator())
-                .setMember(command.getMember())
+                .setUser(cardOwner)
+                .setChat(command.getId())
                 .vBuild();
-        var chatPreview = ChatPreview
+        String chatName;
+        if(command.getCreator().equals(cardOwner)){
+            chatName = command.getMemberName();
```

Comment:
This is the same idea which you previously had in the respective projection. 

Let's move (or extract) it to some production-code-level utility, test it separately, and then use in both `...Projection` implementation and in tests.